examples/k3s/composition.yaml

```diff
@@ -0,0 +1,119 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: k3sclusters.deployments.equinix.com
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  compositeTypeRef:
+    apiVersion: deployments.equinix.com/v1alpha1
+    kind: K3sCluster
+  resources:
+    - name: k3s-project
+      base:
+        apiVersion: metal.equinix.jet.crossplane.io/v1alpha1
+        kind: Project
+      patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: spec.forProvider.name
+    - name: k3s-server
+      base:
+        apiVersion: metal.equinix.jet.crossplane.io/v1alpha1
+        kind: Device
+        metadata:
+          labels:
+            k3s-server: 'true'
+        spec:
+          forProvider:
+            hostname: k3s-server
+            plan: c3.small.x86
+            operatingSystem: ubuntu_20_04
+            billingCycle: hourly
+            locked: false
+            networkType: hybrid
+            tags:
+            - crossplane
+            projectIdSelector:
+              matchControllerRef: true
+          writeConnectionSecretToRef:
+            name: server
+            namespace: crossplane-system
+          reclaimPolicy: Delete
+      patches:
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.accessPublicIpv4
+          toFieldPath: status.serverIP
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.metro
+          toFieldPath: spec.forProvider.metro
+        - type: CombineFromComposite
+          toFieldPath: spec.forProvider.userData
```

Comment:
The `userData` patch is not applying

Reply:
TF `user_data` is sensitive. The field, as defined in the CRD, is `userDataSecretRef`. 

This will have to be done using `writeConnectionSecretToRef`.

Reply:
I see that the script requires only the token as variable. Would it make sense to store the whole script in a `Secret` and only take a reference of that secret in your claim? It will force the claim author to know rest of the content but at least the token won't appear in a resource other than `Secret`.